pkg/controller/utils/datadog/metrics_forwarder.go

```diff
@@ -586,21 +576,23 @@ func (mf *metricsForwarder) getKeyFromSecret(dda *datadoghqv1alpha1.DatadogAgent
 }
 
 // updateStatusIfNeeded updates the Datadog metrics forwarding status in the DatadogAgent status
-func (mf *metricsForwarder) updateStatusIfNeeded(dda *datadoghqv1alpha1.DatadogAgent, err error) error {
+func (mf *metricsForwarder) updateStatusIfNeeded(dda *datadoghqv1alpha1.DatadogAgent, err error) {
 	now := metav1.NewTime(time.Now())
-	newStatus := dda.Status.DeepCopy()
-	condition.UpdateDatadogAgentStatusConditionsFailure(newStatus, now, datadoghqv1alpha1.ConditionTypeDatadogMetricsError, err)
-	if err == nil {
-		condition.UpdateDatadogAgentStatusCondition(newStatus, now, datadoghqv1alpha1.ConditionTypeActiveDatadogMetrics, corev1.ConditionTrue, "Datadog metrics forwarding ok", false)
-	} else {
-		condition.UpdateDatadogAgentStatusCondition(newStatus, now, datadoghqv1alpha1.ConditionTypeActiveDatadogMetrics, corev1.ConditionFalse, "Datadog metrics forwarding error", false)
+	conditionStatus := corev1.ConditionTrue
+	description := "Datadog metrics forwarding ok"
+	if err != nil {
+		conditionStatus = corev1.ConditionFalse
+		description = "Datadog metrics forwarding error"
 	}
-	if !apiequality.Semantic.DeepEqual(&dda.Status, newStatus) {
-		updatedDad := dda.DeepCopy()
-		updatedDad.Status = *newStatus
-		return mf.k8sClient.Status().Update(context.TODO(), updatedDad)
+
+	mf.Lock()
+	defer mf.Unlock()
+	if mf.status == nil {
+		newCondition := condition.NewDatadogAgentStatusCondition(datadoghqv1alpha1.ConditionTypeActiveDatadogMetrics, conditionStatus, now, "", description)
+		mf.status = &newCondition
+	} else {
+		mf.status = condition.UpdateDatadogAgentStatusCondition(mf.status, now, datadoghqv1alpha1.ConditionTypeActiveDatadogMetrics, conditionStatus, description)
```

Comment:
wdyt of using a thread safe `setStatus` method here instead?

Reply:
good idea 👍